src/main/learnocaml_client.ml

```diff
@@ -731,31 +740,50 @@ module Fetch = struct
            (Token.to_string token)
       | e -> Lwt.fail e
 
-  let write_save_files save =
-  Lwt_list.iter_s (fun (id, st) ->
+  let write_save_files lst save =
+    let pred x = match lst with
```

Comment:
This `pred` function seems weird. Can you describe its purpose?

Reply:
I am not good at choosing names :) It is a predicate to decide if we have to get the file or not. If no files where specified (hence an empty `lst`), we have to get all files, so the predicate is always true.

Reply:
Then, we need a comment to explain this.